problems/http.go

```diff
@@ -21,8 +22,18 @@ func Generic(status int) Problem {
 func WriteResponse(ctx context.Context, err error, w http.ResponseWriter, r *http.Request) {
 	problem := FromError(err)
 
-	if decoratableProblem, ok := problem.(RequestDecoratableProblem); ok {
+	// reflect.Value to the underlying implementation of the Problem interface
+	problemValue := reflect.Indirect(reflect.ValueOf(problem))
```

Comment:
Maybe it is possible to extract the decoration part into one function call from within the WriteResponse function